src/resolver.cpp

```diff
@@ -120,7 +120,8 @@ std::string UsdOpenAssetIOResolver::_CreateIdentifier(
     // (for now).
     identifier = assetPath;
   } else {
-    identifier = ArDefaultResolver::_CreateIdentifier(assetPath, anchorAssetPath);
+    identifier = ArDefaultResolver::_CreateIdentifier(
+        assetPath, ArResolvedPath{locationForEntity(anchorAssetPath).value_or(anchorAssetPath)});
```

Comment:
[fix] Redundan]t wrappping in `ArResolvedPath`

Reply:
`_CreateIdentifier` takes `ArResolvedPath`s and cannot implicitly convert from the `std::string` returned by `locationForEntity`.

Reply:
Yeah - this was from before, when it was already wrapped.

Reply:
Continuing on from the other comment on this topic - this is also re-wrapping `anchorAssetPath` in the fallback case. I guess were going to remove this soon anyway, but would it make more sense to just return an Ar path from the lookup?

Reply:
Initially we did return an `ArResolvedPath`, as you noticed, but then we need to unwrap it again in `_GetExtension`, so we went with returning a `std::string` as a the lowest common denominator. We also discussed a possible template-based solution, where we could vary the return type, but decided to keep things simple.

Reply:
Implemented a template-based solution in 4b077603 - can drop that commit if undesired.

Reply:
noice!